### PR TITLE
'Fix' Smashing drive non-nvidia hardware

### PR DESCRIPTION
**Pull request by LukeUsher**
The issue seems to be because non-nvidia hardware clamps the fog values when below 0, making the formula for fog evaluate to (0 - z) / 0, invoking undefined behavior and breaking the scene.

This fix attempts to recover from this situation by taking the absolute value of the fog start/end, in the case when the value is < 0. 

A LOG_TEST_CASE has been added to detect this situation in future, so we can collect test-cases.

Before:
image

After:
image

### CODE REVIEW

File: src/core/hle/D3D8/XbState.cpp [hunk 1]
```diff
@@ -26,7 +26,8 @@
 // ******************************************************************
 #define _XBOXKRNL_DEFEXTRN_
 #define LOG_PREFIX CXBXR_MODULE::D3DST
-
+
+#include "core\kernel\init\CxbxKrnl.h"
```
Comment (LukeUsher) [inline src/core/hle/D3D8/XbState.cpp]: This is required because LOG_TEST_CASE doesn't work without this,

**Comment by Margen67** [comment]
Confirmed working on NVIDIA
image

**Comment by literalmente-game** [comment]
Checked with the games listed below, no regressions found with them:
- Alien Hominid
- Bloodrayne 2
- Crash 'N' Burn
- Dinotopia - The Sunstone Odyssey
- FIFA Street
- Jet Set Radio Future
- Outlaw Golf
- Outrun 2
- Panzer Dragoon Orta
- Psyvariar 2 - Extend Edition
- Star Wars - Jedi Starfighter
- Tennis Master Series 2003

**Comment by LukeUsher** [comment]
Okay, cool, happy to merge based on this

**Pull request merged**